crates/rpc/rpc/src/eth/api/mod.rs

```diff
@@ -370,6 +370,22 @@ where
     }
 }
 
+/// The wrapper type for gas limit
+#[derive(Debug, Clone, Copy)]
+pub struct GasCap(u64);
```

Comment:
this should also have a `Default` impl for the 50M default

Reply:
Should I use the ETHEREUM_BLOCK_GAS_LIMIT or just hardcode the value instead?

Reply:
You could use this:
https://github.com/paradigmxyz/reth/blob/5ad9b32cbcbb9ed1e08a645674731cc78a0f7ed3/crates/rpc/rpc-builder/src/constants.rs#L10-L15

Reply:
I tried using that but I couldn't import the `rpc-builder` crate due to cyclic dependencies occurring. Can you give me some tips about how to solve this?

Reply:
In that case I would move the `RPC_DEFAULT_GAS_CAP` from `rpc/rpc-builder/src/constants.rs` to `rpc/rpc/eth/api/mod.rs` maybe? And then it should b exposed in `reth-rpc`, and imported wherever it's used in `rpc-builder`

Reply:
Done!